File: mavkit-python/examples/params_roundtrip.py
"""Download all parameters and round-trip through the param file format."""

import asyncio
import os

import mavkit


async def main():
    bind_addr = os.environ.get("MAVKIT_EXAMPLE_UDP_BIND", "0.0.0.0:14550")

    vehicle = await mavkit.Vehicle.connect_udp(bind_addr)
    params = vehicle.params()
    download = params.download_all()

    store = await download.wait()
    state = params.latest()
    print(
        f"downloaded {len(store)} params (sync={state.sync if state is not None else None})"
    )

    text = mavkit.format_param_file(store)
    parsed = mavkit.parse_param_file(text)
    print(f"formatted+parsed {len(parsed)} params")

    await vehicle.disconnect()


asyncio.run(main())

File: examples/params_roundtrip.rs
use mavkit::{Vehicle, format_param_file, parse_param_file};

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let bind_addr =
        std::env::var("MAVKIT_EXAMPLE_UDP_BIND").unwrap_or_else(|_| "0.0.0.0:14550".to_string());

    let vehicle = Vehicle::connect_udp(&bind_addr).await?;
    let params = vehicle.params();

    let store = params.download_all()?.wait().await?;
    println!("downloaded {} params", store.params.len());

    let text = format_param_file(&store);
    let parsed = parse_param_file(&text)?;
    println!("formatted+parsed {} params", parsed.len());

    vehicle.disconnect().await?;
    Ok(())
}
